Content.Server/GameTicking/GameTicker.GamePreset.cs

```diff
@@ -274,6 +275,10 @@ public bool OnGhostAttempt(EntityUid mindId, bool canReturnGlobal, bool viaComma
                 }
             }
 
+            // Free the sentient artifact ghost role when ghosting
+            if(canReturnGlobal && TryComp(playerEntity, out ArtifactComponent? artifact))
+                canReturn = false;
+
```

Comment:
Artifacts shouldn't be hardcoded into ghost code as it just becomes a big mess for forks, probably put in a bool on the component instead or use an event.